### PR TITLE
[TE] Bug fix for user report to add function Id

### PR DESCRIPTION
**Pull request by cecilynie**
User report anomalies to be exposed to UI
Add functionId to user report anomalies to let API fetch anomalies under function Id.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java [hunk 1]
```diff
@@ -1011,7 +1012,8 @@ private String getDimensions(String dataset, String exploreDimensions) throws Ex
 
       for (MergedAnomalyResultDTO mergedAnomaly : mergedResults) {
         // if use notified flag, only keep anomalies isNotified == true
-        if ( (useNotified && mergedAnomaly.isNotified()) || !useNotified) {
+        if ( (useNotified && mergedAnomaly.isNotified()) || !useNotified || mergedAnomaly.getAnomalyResultSource().equals(
```
Comment (cyy0714) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java]: Use AnomalyResultSource.USER_LABELED_ANOMALY).equals(...) instead.

**Comment by cecilynie** [comment]
@apucher It's because FE is fetching anomalies using anomaly function Id. The create user report logic  is adding anomaly function but not adding the function Id. From integration test I see the created anomaly has anomaly functionid=0. This is a quick fix. A better fix should be when adding "anomaly function", the functionId field should also be populated.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java (hunk 1 above)
Comment (cecilynie) [reply thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/AnomalyResource.java]: fixed. thanks for advising!

**Comment by apucher** [comment]
I'm referring to the change in the if-statement on line 1014

**Comment by cecilynie** [comment]
@apucher "User-Report" Anomaly is "not notified". Frontend is using "notified flag" to show the anomaly list, hence previously "User-Report" is never shown in the list.

**Pull request merged**